src/Microsoft.ML.TimeSeries/SrCnnEntireTransformer.cs

```diff
@@ -0,0 +1,903 @@
+                : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
+            {
+                if (!inputSchema.TryGetColumnIndex(parent.InputColumnName, out _inputColumnIndex))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", parent.InputColumnName);
+
+                var colType = inputSchema[_inputColumnIndex].Type;
+                if (!(colType is SrCnnTsPointDataViewType))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", parent.InputColumnName, "SrCnnTsPoint", colType.ToString());
+
+                _parent = parent;
+                if (_parent.OutputLength == 3)
+                {
+                    _slotNames = new VBuffer<ReadOnlyMemory<char>>(_parent.OutputLength, new[] { "Is Anomaly".AsMemory(), "Raw Score".AsMemory(), "Mag".AsMemory() });
+                }
+                else if (_parent.OutputLength == 7)
+                {
+                    _slotNames = new VBuffer<ReadOnlyMemory<char>>(_parent.OutputLength, new[] { "Is Anomaly".AsMemory(), "Anomaly Score".AsMemory(), "Mag".AsMemory(),
+                        "Expected Value".AsMemory(), "Boundary Unit".AsMemory(), "Upper Boundary".AsMemory(), "Lower Boundary".AsMemory() });
+                }
+                else if (_parent.OutputLength == 4)
+                {
+                    _slotNames = new VBuffer<ReadOnlyMemory<char>>(_parent.OutputLength, new[] { "Is Anomaly".AsMemory(), "Raw Score".AsMemory(), "Mag".AsMemory(),
+                        "Expected Value".AsMemory() });
+                }
+            }
+
+            protected override DataViewSchema.DetachedColumn[] GetOutputColumnsCore()
+            {
+                var meta = new DataViewSchema.Annotations.Builder();
+                meta.AddSlotNames(_parent.OutputLength, GetSlotNames);
+                var info = new DataViewSchema.DetachedColumn[1];
+                info[0] = new DataViewSchema.DetachedColumn(_parent.OutputColumnName, new VectorDataViewType(NumberDataViewType.Double, _parent.OutputLength), meta.ToAnnotations());
+                return info;
+            }
+
+            public void GetSlotNames(ref VBuffer<ReadOnlyMemory<char>> dst) => _slotNames.CopyTo(ref dst, 0, _parent.OutputLength);
+
+            protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, bool> activeOutput, out Action disposer)
+            {
+                Contracts.AssertValue(input);
+                Contracts.Assert(0 <= iinfo && iinfo < _parent.ColumnPairs.Length);
+
+                var src = default(SrCnnTsPoint);
+                var getSrc = input.GetGetter<SrCnnTsPoint>(input.Schema[ColMapNewToOld[iinfo]]);
+
+                disposer = null;
+
+                ValueGetter<VBuffer<double>> del =
+                    (ref VBuffer<double> dst) =>
+                    {
+                        getSrc(ref src);
+                        if (src == null)
+                            return;
```

Comment:
>return [](start = 28, length = 6)

What does `src == null` indicate? Don't you want to assign something to `dst` so that users would know that `src` was `null`? (maybe all zeros? or all `NaN`s?)

Reply:
> > return [](start = 28, length = 6)
> 
> What does `src == null` indicate? Don't you want to assign something to `dst` so that users would know that `src` was `null`? (maybe all zeros? or all `NaN`s?)

Updated. Assign NaN now.